py/desitarget/gfa.py

```diff
@@ -218,6 +218,36 @@ def add_gfa_info_to_fa_tiles(gfa_file_path="./", fa_file_path=None, output_path=
             fitsio.write(tileout, gfa_data, extname='GFA')
 
 
+def gaia_morph(gaia):
+    """Assign morphological type to Gaia point sources.
+
+    Parameters
+    ----------
+    gaia: :class:`~numpy.ndarray`
+        Numpy structured array containing at least the columns `TYPE`,
+        `GAIA_PHOT_G_MEAN_MAG` and `GAIA_ASTROMETRIC_EXCESS_NOISE`.
+
+    Returns
+    -------
+    :class:`~numpy.ndarray`
+        Input array, with the ``TYPE`` column updated to "GPSF" or "GGAL"
+        based on a morphological cut with Gaia.
+    """
```

Comment:
This appears to modify the original input array, which is not clear from the docstring.

It would be useful to have a way to update the TYPE column only if it was not already set to something useful from the legacy surveys.  Consider having this function just return the array of GPSF/GGAL gaia types and leave it to the user to chose which rows of the original table to update.

Reply:
I update the `gaia_morph` function to simply return an array of `GGAL` or `GPSF`, which should satisfy this suggestion. I then used that output elsewhere in the code to set the morphological type for Gaia-only sources.